Lint session fails with flake8==6.1.0

`flake8==6.1.0` was released on July 29th. 
https://pypi.org/project/flake8/6.1.0/#history

When running the lint check with `flake8==6.1.0` , the check fails with the following errors:

```
(py38) partheniou@partheniou-vm-3:~/git/python-storage$ nox -s lint
nox > Running session lint
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/lint
nox > python -m pip install flake8 black==22.3.0
nox > black --check docs google tests noxfile.py setup.py
All done! ✨ 🍰 ✨
56 files would be left unchanged.
nox > flake8 google tests
google/cloud/storage/acl.py:344:5: F811 redefinition of unused 'client' from line 140
google/cloud/storage/client.py:150:16: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
tests/unit/test_transfer_manager.py:195:16: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
nox > Command flake8 google tests failed with exit code 1
nox > Session lint failed.
```

--- a/google/cloud/storage/acl.py
+++ b/google/cloud/storage/acl.py
@@ -137,7 +137,6 @@ class ACL(object):
 
     # Subclasses must override to provide these attributes (typically,
     # as properties).
-    client = None
     reload_path = None
     save_path = None
     user_project = None

--- a/google/cloud/storage/client.py
+++ b/google/cloud/storage/client.py
@@ -147,7 +147,7 @@ class Client(ClientWithProject):
         kw_args["api_endpoint"] = storage_host if _is_emulator_set else None
 
         if client_options:
-            if type(client_options) == dict:
+            if isinstance(client_options, dict):
                 client_options = google.api_core.client_options.from_dict(
                     client_options
                 )

--- a/tests/unit/test_transfer_manager.py
+++ b/tests/unit/test_transfer_manager.py
@@ -192,7 +192,7 @@ def test_upload_many_suppresses_412_with_skip_if_exists():
         worker_type=transfer_manager.THREAD,
     )
     for result in results:
-        assert type(result) == exceptions.PreconditionFailed
+        assert isinstance(result, exceptions.PreconditionFailed)
 
 
 def test_upload_many_with_processes():

